shardingsphere-test/shardingsphere-integration-test/shardingsphere-integration-test-scaling/src/test/java/org/apache/shardingsphere/integration/data/pipeline/cases/base/BaseITCase.java

```diff
@@ -317,5 +318,44 @@ protected void assertCheckMatchConsistencySuccess() throws InterruptedException
         assertThat(previewResults.stream().map(each -> each.get("data_source_name")).collect(Collectors.toSet()), is(new HashSet<>(Arrays.asList("ds_2", "ds_3", "ds_4"))));
         assertThat(targetSources, is(new HashSet<>(Arrays.asList("SELECT COUNT(1) FROM t_order_0 UNION ALL SELECT COUNT(1) FROM t_order_3",
                 "SELECT COUNT(1) FROM t_order_1 UNION ALL SELECT COUNT(1) FROM t_order_4", "SELECT COUNT(1) FROM t_order_2 UNION ALL SELECT COUNT(1) FROM t_order_5"))));
+        restoreScalingSourceWriting(jobId);
+    }
+    
+    protected void restoreScalingSourceWriting(final String jobId) {
+        executeWithLog(String.format("RESTORE SCALING SOURCE WRITING %s", jobId));
+    }
+    
+    protected abstract void assertStopScalingSourceWriting();
+    
+    protected abstract void assertRestoreScalingSourceWriting();
+    
+    protected boolean executeUpdate(final String sql) {
+        log.info("jdbcTemplate execute:{}", sql);
+        try {
+            jdbcTemplate.update(sql);
+            return true;
+            // CHECKSTYLE:OFF
+        } catch (final Exception ex) {
+            // CHECKSTYLE:ON
+            assertSQLException((SQLException) ex.getCause());
+            return false;
+        }
+    }
+    
+    protected boolean executeDDL(final String sql) {
+        log.info("jdbcTemplate execute:{}", sql);
+        try {
+            jdbcTemplate.execute(sql);
+            return true;
+            // CHECKSTYLE:OFF
+        } catch (final Exception ex) {
+            // CHECKSTYLE:ON
+            assertSQLException((SQLException) ex.getCause());
+            return false;
+        }
+    }
+    
+    private void assertSQLException(final SQLException exception) {
+        assertTrue(exception.getMessage().endsWith("The database sharding_db is read-only"));
     }
```

Comment:
executeUpdate / executeDDL / assertSQLException is not common enough, it could be merged into assertRestoreScalingSourceWriting.

Reply:
OK， I will try

Reply:
Fixed